onnxruntime/core/framework/prepacked_weights_cache.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/common.h"
+#include "core/framework/allocatormgr.h"
+#include "core/framework/tensor.h"
+
+namespace onnxruntime {
+
+struct PackedWeight {
+  // Some weights may be associated with multiple pre-packed buffers.
+  // Hence we hold them in containers. It is upto the dev implementing each PrePack()
+  // method to define what gets stored in which position.
+  std::vector<BufferUniquePtr> buffers_;
+  std::vector<size_t> weights_sizes_;
+  std::vector<TensorShape> shapes_;
+  bool has_cached_ = false;
+};
+
+// TODO: Make this class thread-safe ?
+class PrepackedWeightsCache {
+ public:
+  PrepackedWeightsCache() {
+  }
+
+  ~PrepackedWeightsCache() = default;
+
+  AllocatorPtr GetAllocator(const char* device_name) {
+    auto name = std::string(device_name);
```

Comment:
Even more robust and generic would be `std::string_view` (C++17), because then you don't need to worry about whether the string is already null terminated (it's allowed to be a subslice directly, without allocating a separate string buffer just to append a null), and you don't need to dynamically allocate an `std::string` just to pass the parameter in the cases that the source wasn't already an `std::string` (most of the time in ORT it already was an `std::string` anyway, but `std::string_view` offers lightweight flexibility and generalizes the interface more elegantly than either `string` or raw `char*` can).

Reply:
We cannot use C++17 yet.